static/js/src/renewal-modal.js

```diff
@@ -1,469 +1,708 @@
-  const cardExpiryEl = modal.querySelector(".js-customer-card-expiry");
-  const cardImgEl = modal.querySelector(".js-customer-card-brand");
-  const cardTextEl = modal.querySelector(".js-customer-card");
-  const customerEmailEl = modal.querySelector(".js-customer-email");
-  const customerNameEl = modal.querySelector(".js-customer-name");
-  const nameElement = modal.querySelector(".js-renewal-name");
-  const processPaymentButton = modal.querySelector(".js-process-payment");
-  const quantityElement = modal.querySelector(".js-renewal-quantity");
-  const startElement = modal.querySelector(".js-renewal-start");
-  const totalElement = modal.querySelector(".js-renewal-total");
-
-  const changePaymentMethodButton = modal.querySelector(
-    ".js-change-payment-method"
-  );
-  const cancelModalButton = modal.querySelector(".js-cancel-modal");
-
-  const cardErrorElement = document.getElementById("card-errors");
-  const renewalErrorElement = document.getElementById("renewal-errors");
-
-  // initialise Stripe
-  const stripe = Stripe("pk_test_yndN9H0GcJffPe0W58Nm64cM00riYG4N46");
-
-  // customise the Stripe card field
-  const style = {
-    base: {
-      iconColor: "#e95420",
-      color: "#111",
-      fontWeight: 300,
-      fontFamily: "Ubuntu, sans-serif",
-      fontSmoothing: "antialiased",
-      fontSize: "18px",
-      "::placeholder": {
-        color: "#666",
-      },
-      ":-webkit-autofill": {
-        color: "#666",
-      },
-    },
-  };
+import { parseForErrorObject } from "./stripe/error-handler.js";
 
-  // create the Stripe card input, and apply the style to it
-  const elements = stripe.elements();
-  const card = elements.create("card", { style });
+import {
+  setPaymentInformation,
+  setRenewalInformation,
+} from "./stripe/set-modal-info.js";
+
+const modal = document.getElementById("renewal-modal");
+
+const form = document.getElementById("details-form");
+const errorDialog = document.getElementById("payment-error-dialog");
+const progressIndicator = document.getElementById("js-progress-indicator");
+
+const addPaymentMethodButton = modal.querySelector(".js-payment-method");
+const processPaymentButton = modal.querySelector(".js-process-payment");
+const changePaymentMethodButton = modal.querySelector(
+  ".js-change-payment-method"
+);
+const cancelModalButton = modal.querySelector(".js-cancel-modal");
+const closeModalButton = modal.querySelector(".js-close-modal");
+
+const cardErrorElement = document.getElementById("card-errors");
+const renewalErrorElement = document.getElementById("renewal-errors");
+
+const vatCountries = [
+  "AL",
+  "AT",
+  "BE",
+  "BG",
+  "HR",
+  "CY",
+  "CZ",
+  "DK",
+  "EE",
+  "FI",
+  "FR",
+  "DE",
+  "EL",
+  "HU",
+  "IE",
+  "IT",
+  "LV",
+  "LT",
+  "LU",
+  "MT",
+  "NL",
+  "PL",
+  "PT",
+  "RO",
+  "SK",
+  "SI",
+  "ES",
+  "SE",
+  "GB",
+];
+
+// initialise Stripe
+const stripe = window.Stripe("pk_test_yndN9H0GcJffPe0W58Nm64cM00riYG4N46");
```

Comment:
Should this key be in a config file? I imagine there is a different one for production?